test_workspace/Queue.bosatsu

```diff
@@ -0,0 +1,107 @@
+package Queue
+
+export [ Queue,
+  empty_Queue, fold_Queue, push, pop, pop_value, pop_Queue, reverse_Queue, eq_Queue,
+  queue_to_List,
+]
+
+struct Queue(front: List[a], back: List[a])
+
+empty_Queue: forall a. Queue[a] = Queue([], [])
+
+# convenient local alias
+empty = empty_Queue
+
+def queue_from_List(list: List[a]) -> Queue[a]:
+  Queue(list, [])
+
+def push(queue: Queue[a], item: a) -> Queue[a]:
+  Queue(f, b) = queue
+  Queue(f, [item, *b])
+
+def pop(queue: Queue[a]) -> Option[(a, Queue[a])]:
+  match queue:
+    Queue([h, *t], b): Some((h, Queue(t, b)))
+    Queue([], b):
+      match b.reverse:
+        []: None
+        [h, *t]: Some((h, Queue(t, [])))
+
+def pop_value(queue: Queue[a]) -> Option[a]:
+  match pop(queue):
+    Some((a, _)): Some(a)
+    None: None
+
+# drop an item off and return the rest, or empty
+def pop_Queue(queue: Queue[a]) -> Queue[a]:
```

Comment:
it kind of feels like your convention is inconsistent here. In other places the `_Type` suffix means "this is the version of this function for `Type`" but here you're saying "this returns the `Queue` side of the tuple". Maybe the convention isn't that big of a deal